planning/mission_planner/src/lanelet2_plugins/default_planner.cpp

```diff
@@ -255,15 +355,11 @@ PlannerPlugin::HADMapRoute DefaultPlanner::plan(const RoutePoints & points)
   HADMapRoute route_msg;
   RouteSections route_sections;
 
-  if (!is_goal_valid(points.back())) {
-    RCLCPP_WARN(logger, "Goal is not valid! Please check position and angle of goal_pose");
-    return route_msg;
-  }
-
+  lanelet::ConstLanelets path_lanelets;
   for (std::size_t i = 1; i < points.size(); i++) {
     const auto start_check_point = points.at(i - 1);
     const auto goal_check_point = points.at(i);
-    lanelet::ConstLanelets path_lanelets;
+    path_lanelets.clear();
```

Comment:
I feel this implementation had better be clean.

How about reverting this part of the change and give goal's lanelet to is_goal_valid.
Then is_goal_valid search previous lanelets if the footprint is inside the lanelets the same way as searching next lanelets.

Let me know if this idea does not sound good to you. I 'll explain more.

Reply:
If we don't want to use recursive call for previous lanes as we talked about before, it might be good to use path_lanelets for previous lanelets.

But in order to not affect the checkpoint mechanism, I made a small update here in the last commit.